documentation/modules/ROOT/pages/02-architecture.adoc

```diff
@@ -1,104 +1,255 @@
 
 [#tech_stack]
 == Technology Stack
 
-// Change links and text here as you see fit.
+Here's the list of technologies used in this solution and its examples:
+
 * Red Hat supported products
 ** https://www.redhat.com/en/technologies/cloud-computing/openshift[Red Hat OpenShift]
-** Red Hat Application Foundation
-*** https://access.redhat.com/products/quarkus[Quarkus]
-*** https://www.redhat.com/en/technologies/cloud-computing/openshift/openshift-streams-for-apache-kafka[Kafka Streams]
-* Other open source products:
-** https://www.postgresql.org/[PostgreSQL database]
-** https://helm.sh/[Helm]
-
+&mdash; Orchestrate containerized applications.
+Based on https://kubernetes.io/[Kuberentes].
+** https://www.redhat.com/en/technologies/cloud-computing/openshift/serverless[Red Hat OpenShift Serverless]
+&mdash; Provides the _Event Mesh_ and _Serverless_ capabilities.
+Based on https://knative.dev[Knative] project.
+** https://swc.saas.ibm.com/en-us/redhat-marketplace/products/red-hat-amq[Red Hat AMQ Streams]
+&mdash; (Optional) Provides a persistence for _Event Mesh_, likely needed in production.
+Based on https://strimzi.io/[Strimzi] project.
+* Other open source technologies:
+** https://cloudevents.io/[CloudEvents] &mdash; Provides a standard for event metadata
+** Rust and https://access.redhat.com/products/quarkus[Quarkus] &mdash; Implementation examples.
+** https://opentelemetry.io/[OpenTelemetry] &mdash; (Optional) Facilitates tracing for observability.
 
 [#in_depth]
 == An in-depth look at the solution's architecture
 
-Technical description including all or some of the following: architecture ir ed diagrams. In-depth details of the decisions made and solutions used. Description of each service and what it is used for. Description of any integration.
+Building applications around the Event Mesh is a solution that can be applied both to existing and new projects.
+For existing applications, domain boundaries can guide the division into modular components, which may evolve into separate services.
+These modules will generate commands intended for the _Event Mesh_, which will then route these events to their designated endpoints.
+The _Event Mesh_ will allow for such transition, gradually making the system not only more responsive but also better suited to the real-world business logic.
+
+=== Problem Analysis
+
+Traditional systems often enforce strict transactional consistency, which can impede application performance and compromise business logic.
+For instance, upon the completion of a ride-sharing service at the end-user's destination, the system should reliably capture this real-world event.
+The capture should be performed regardless of any potential operational disruptions affecting dependent services (e.g., invoicing).
+
+In such scenarios, transactional applications typically return an error, which prevents any data from being changed, and causes the loss of real-world intent of end-users.
```

Comment:
You could add here also, that distributed transactions typically embrace a number of steps in a user workflow which are all undone, causing frustration for users who have to redo everything, not ally the part that didn't work. (just a suggestion)

Reply:
Yes, this was my intention. And I think I wrote about it somewhere, but later maybe removed it. I agree, this could be a good idea to explain the problem a bit more clearly!

Reply:
Found it. I know I wrote about it somewhere.

It is on the demo page, here: https://github.com/cardil/solution-pattern-event-mesh-for-microservices/pull/1/files#diff-460e13111de481b79a5cdc6f0c893cc3c708d0dbca44893dd231fd6186bb7801R73

Maybe in fact it would be better to move it out of that page and put it here...